x-pack/platform/plugins/shared/observability_ai_assistant/server/prompts/system_prompt.ts

```diff
@@ -0,0 +1,290 @@
+          *   **DO NOT** generate, explain, or correct ES|QL queries yourself. Always delegate to the \`${QUERY_FUNCTION_NAME}\` tool, even if it was just used or if it previously failed.
+          ${
+            datasetTools.length
+              ? `*   If ${datasetTools.join(
+                  ' or '
+                )} return no results, but the user asks for a query, *still* call the \`${QUERY_FUNCTION_NAME}\` tool to generate an *example* query based on the request.`
+              : ''
+          }
+      5. When a user requests paginated results using ES|QL (e.g., asking for a specific page or part of the results), you must inform them that ES|QL does not support pagination or offset. Clearly explain that only limiting the number of results (using LIMIT) is possible, and provide an example query that returns the first N results. Do not attempt to simulate pagination or suggest unsupported features. Always clarify this limitation in your response.
+      6. When converting queries from another language (e.g SPL, LogQL, DQL) to ES|QL, generate functionally equivalent ES|QL query using the available index and field information. Infer the indices and fields from the user's query and always call \`${QUERY_FUNCTION_NAME}\` tool to provide a **valid and functionally equivalent** example ES|QL query. Always clarify any field name assumptions and prompt the user for clarification if necessary.
+      7. **Critical ES|QL syntax rules:**
+          * When using \`DATE_FORMAT\`, any literal text in the format string **MUST** be in single quotes. Example: \`DATE_FORMAT("d 'of' MMMM yyyy", @timestamp)\`.
+          * When grouping with \`STATS\`, use the field name directly. Example: \`STATS count = COUNT(*) BY destination.domain\`.
+      `)
+    );
+  }
+
+  // Section Four: Tool Usage Guidelines
+  if (availableFunctionNames.length) {
+    const usage: string[] = [];
+
+    if (toolsWithTimeRange.length) {
+      usage.push(
+        `**Time Range Handling:** As stated in Core Principles, for tools requiring time ranges${toolsWithTimeRange.join(
+          ', '
+        )}, ${
+          isFunctionAvailable(CONTEXT_FUNCTION_NAME)
+            ? `first try \`${CONTEXT_FUNCTION_NAME}\` to find time range. If no time range is found in context,`
+            : ''
+        } use the default (\`start='now-15m'\`, \`end='now'\`) and inform the user.`
+      );
+    }
+
+    if (isFunctionAvailable(GET_DATASET_INFO_FUNCTION_NAME)) {
+      usage.push(
+        `**Get Dataset Information:** Use the \`${GET_DATASET_INFO_FUNCTION_NAME}\` tool to get information about indices/datasets available and the fields available on them. 
+         * Providing an empty string as index name will retrieve all indices. Otherwise list of all fields for the given index will be given. 
```

Comment:
@SrdjanLL how about?

```suggestion
         * Providing an empty string as index name will retrieve all indices. Otherwise list of all fields for the given index will be returned. 
```